src/lib/pubkey/pk_algs.cpp

```diff
@@ -346,47 +325,82 @@ create_private_key(const std::string& alg_name,
 #if defined(BOTAN_HAS_RSA)
    if(alg_name == "RSA")
       {
-      const size_t rsa_bits = (params.empty() ? 3072 : to_u32bit(params));
-      return std::make_unique<RSA_PrivateKey>(rng, rsa_bits);
+      const size_t modulus_bits = [&]() -> size_t
+         {
+         if(params.empty())
+            return 3072;
+         return to_u32bit(params);
+         }();
+
+      return std::make_unique<RSA_PrivateKey>(rng, modulus_bits);
```

Comment:
Out of curiosity: What was wrong with the ternary?

Reply:
You know I think here actually the ternary worked ok and I just autopiloted the conversion after I did the other cases. I'll revert this bit.

Reply:
Reverted